gcloud-java-dns/README.md

```diff
@@ -0,0 +1,324 @@
+credentials. You can then make API calls by calling methods on the DNS service object. The simplest
+way to authenticate is to use [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
+These credentials are automatically inferred from your environment, so you only need the following
+code to create your service object:
+
+```java
+import com.google.gcloud.dns.Dns;
+import com.google.gcloud.dns.DnsOptions;
+
+Dns dns = DnsOptions.defaultInstance().service();
+```
+
+For other authentication options, see the [Authentication](https://github.com/GoogleCloudPlatform/gcloud-java#authentication) page.
+
+#### Managing Zones
+DNS records in `gcloud-java-dns` are managed inside containers called "zones". `ZoneInfo` is a class
+which encapsulates metadata that describe a zone in Google Cloud DNS, but it cannot interact with
+the Cloud DNS service on its own. `Zone`, a subclass of `ZoneInfo`, adds service-related
+functionality over `ZoneInfo`. In this code snippet, we create a new zone in which we intend to
+manage DNS record for domain `someexampledomain.com.`
+
+Add the following imports at the top of your file:
+
+```java
+import com.google.gcloud.storage.Zone;
+import com.google.gcloud.storage.ZoneInfo;
+```
+
+Then add the following code to create a zone.
+
+*Important: Zone names within each project have to be unique. If you choose a zone name that already
+exists, you'll get a helpful error message telling you to choose another name. In the code below,
+replace "my_unique_zone" with a unique zone name. See more about naming rules [here](https://cloud.google.com/dns/api/v1/managedZones).*
+
+*Important: The service may require that you verify ownership of the domain name that you are
+creating a zone for. Hence, we recommend that you do so beforehand. You can verify ownership of
+a domain name [here](https://www.google.com/webmasters/verification/home). Note that `gcloud-java`
+works with fully qualified domain names which must end with a period.*
+
+```java
+// Create a zone metadata
+String zoneName = "my_unique_zone"; // Change this zone name which is unique within your project
+String domainName = "someexampledomain.com."; // Change this to a domain which you own
+String description = "This is a gcloud-java-dns sample zone.";
+ZoneInfo zoneInfo = ZoneInfo.of(zoneName, domainName, description);
+
+// Create zone in Google Cloud DNS
+Zone zone = dns.create(zoneInfo);
+```
+
+You now have an empty zone hosted in Google Cloud DNS which is ready to be populated with DNS
+records for domain name `someexampledomain.com.` Upon creating the zone, the cloud service
+automatically assigned a set of DNS servers to this zone which will be hosting the records, and
+created the default required SOA and NS records for the domain inside this zone. You can read the
+list of assigned DNS servers using accessor method `zone.nameServers()` and for instance print them
```

Comment:
I would change this sentence to "The following snippet reads and prints the list of assigned servers." and put the import on a separate line (like you did for java.util.Iterator) below, just for consistency (I missed that import when skimming through earlier).